lua/randomat2/events/sv_poker.lua

```diff
@@ -0,0 +1,1136 @@
+                
+                self:BeginBetting(nextPly)
+            end
+        end
+    elseif bet < BettingStatus.CHECK then
+        -- Out of sync player fold, used primarily for player disconnecting/death
+        PlayerFolds(ply)
+
+        if not EnoughPlayersRemaining() then
+            if CanDispenseWinnings() then
+                self:CalculateWinner()
+            else
+                net.Start("DeclareNoWinner")
+                net.Broadcast()
+
+                timer.Simple(GetDynamicRoundTimerValue("RoundStateMessage"), function()
+                    self:End()
+                end)
+            end
+        end
+    end
+end
+
+function EVENT:EndBetting()
+    timer.Simple(GetDynamicRoundTimerValue("RoundStateMessage"), function()
+        local epr = EnoughPlayersRemaining() -- This function needs to be ran BEFORE changing player's Status property
+
+        for _, ply in ipairs(self.Players) do
+            if ply.Status ~= BettingStatus.FOLD then
+                ply.Status = BettingStatus.NONE
+            end
+        end
+
+        if self.HaveDiscarded or not epr then
+            self:CalculateWinner()
+        else
+            self:BeginDiscarding()
+        end
+    end)
+end
+
+function EVENT:BeginSecoundRoundBetting()
+    if not self.Started then self:End() return end
+
+    if AllPlayersMatchingBets(true) and IsAllIn(GetHighestBet())then
+        self:CalculateWinner()
+    else
+        self:BeginBetting()
+    end
+end
+
+function EVENT:BeginDiscarding()
+    if not self.Started then self:End() return end
+
+    net.Start("StartDiscard")
+    net.Broadcast()
+
+    self.AcceptingDiscards = true
+
+    timer.Create("AcceptDiscards", GetDynamicRoundTimerValue("RoundStateDiscarding"), 1, function()
+        EVENT_REF:CompletePlayerDiscarding()
+    end)
+end
+
+local function AllPlayersDiscarded()
+    for _, ply in ipairs(EVENT_REF.Players) do
+        if not ply.HasDiscarded then
+            return false
+        end
+    end
+
+    return true
+end
+
+function EVENT:RegisterPlayerDiscard(ply, discardsTable)
+    if not self.Started then self:End() return end
+
+    if not self.AcceptingDiscards then return end
+
+    for _, cardToRemove in ipairs(discardsTable) do
+        local toBeRemoved
+        for index, cardInHand in ipairs(ply.Cards) do
+            if cardToRemove.Rank == cardInHand.Rank and cardToRemove.Suit == cardInHand.Suit then
+                toBeRemoved = index
+            end
+        end
+
+        if toBeRemoved then
+            table.remove(ply.Cards, toBeRemoved)
+        end
+    end
+
+    ply.HasDiscarded = true
+    if AllPlayersDiscarded() then
+        self:CompletePlayerDiscarding()
+    end
+end
+
+function EVENT:CompletePlayerDiscarding()
+    timer.Remove("AcceptDiscards")
+    self.AcceptingDiscards = false
+    self.HaveDiscarded = true
+
+    self:DealDeck(true)
+
+    timer.Simple(GetDynamicRoundTimerValue("RoundStateMessage"), function()
```

Comment:
Recommend against using simple timers because they are not stoppable if, for instance, the event is ended early.